### PR TITLE
feat: basic monitoring of cache + rds via cloud watch

### PR DESCRIPTION
**Pull request by Ardiea**
## Description
This pull request creates default monitors for RDS + elasticache deployments to provide some basic monitoring, mostly focused around load / capacity management. Monitoring is enabled by default and will be added to stacks as they are re-upped. Monitoring can be explicitly disabled with a new configuration arg to OLAmazonCache and OLDatabase config options. 

## How Has This Been Tested?
All monitoring is currently configured as 'warning' meaning it will only page during business hours, not at night. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/ol_infrastructure/components/aws/cloudwatch.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    comparison_operator: str
+    datapoints_to_alarm: PositiveInt = 1
+    description: str
+    enabled: bool = True
+    evaluation_periods: PositiveInt = 2
+    level: str = "warning"
+    metric_name: str
+    name: str
+    namespace: str
+    period: PositiveInt = PositiveInt(300)  # Five minutes
```
Comment (blarghmatey) [inline src/ol_infrastructure/components/aws/cloudwatch.py:26]:
```suggestion
    period: PositiveInt = 300  # Five minutes
```

File: src/ol_infrastructure/components/aws/cloudwatch.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        if statistic not in valid_statistics:
+            raise ValueError(
+                f"statistic: {statistic} is not valid. Valid statistics "
+                "are: {valid_statistics}"
+            )
+        return statistic
+
+    @validator("unit")
+    def is_valid_unit(cls, unit: str) -> Union[str, None]:  # noqa: N805
+        valid_units = [
```
[the remark this answers is not in the document]
Comment (Ardiea) [reply src/ol_infrastructure/components/aws/cloudwatch.py:85]: Today I Learned tuples are immutable. Makes sense.

File: src/ol_infrastructure/infrastructure/monitoring/__main__.py [hunk 3]
```diff
@@ -26,13 +67,37 @@
             bucket_key_enabled=True,
         )
     ),
+    tags=aws_config.merged_tags(
+        {"OU": "open-courseware", "Name": fastly_logging_bucket_name}
```
[the remark this answers is not in the document]
Comment (Ardiea) [reply src/ol_infrastructure/infrastructure/monitoring/__main__.py:71]: Good point. 👍

File: src/ol_infrastructure/components/aws/cloudwatch.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/ol_infrastructure/components/aws/cloudwatch.py:85]: Another option would be to move these units into an `Enum` which can then be used as the type annotation in the config model, which would then enforce membership automatically without requiring the validation method. That has the added benefit of allowing the consumer to use the Enum for populating the parameter, eliminating issues with stringly typed entries.

**Pull request merged**